docs/decisions/20240103-use-jwt-claims-for-agent-admin-auth.md

```diff
@@ -0,0 +1,155 @@
+# Use JWT claims for agent admin access control
+
+- Status: accepted
+- Deciders: Pat Losoponkul, Yurii Shynbuiev, David Poltorak, Shailesh Patil
+- Date: 2024-01-03
+- Tags: multitenancy, authorisation, authentication
+
+Technical Story: [Allow agent admin role to be authenticated by Keycloak JWT | https://input-output.atlassian.net/browse/ATL-6074]
+
+## Context and Problem Statement
+
+External IAM authentication and authorization are currently limited to tenants using the wallets.
+Administrators currently rely on a static API key configured in the agent.
+Enhancing the security of admin access requires extending support for admin authentication through an external IAM.
+This introduces a concept of role to the authentication and authorization process.
+
+The existing tenant authentication and authorization model relies on UMA (user-managed access)
+which defines resources, policies, and permissions.
+Tenants gain access to resources based on defined permissions, effectively controlling resource access.
+In addition to wallet usage, the agent also handles wallet management,
+a functionality utilized by both tenants and administrators.
+While administrators don't directly use the wallet, they oversee its management.
```

Comment:
Should we clarify if the `tenant` and `administrator` roles must be wholly segregated (agent admin user must not own the wallet) to make the system more compliant? 
As a use case, I don't want another user who has a wallet to have admin rights on my wallet.

Reply:
That is a good point. In practice, it is impossible to enforce how keycloak is configured. A user with admin role claim can also have permission to a resource. At the very least, agent auth layer should enforce this check. So if an admin role is present, the `WalletAccessContext` must not be populated. I will update the writing.